apps/tomcat.go

```diff
@@ -0,0 +1,134 @@
+					// Sample line: org.apache.catalina.LifecycleException: An invalid Lifecycle transition was attempted ([before_destroy]) for component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/examples]] in state [STARTED]
+					// Sample line:         at org.apache.catalina.util.LifecycleBase.invalidTransition(LifecycleBase.java:430)
+					// Sample line:         at org.apache.catalina.util.LifecycleBase.destroy(LifecycleBase.java:316)
+					Regex: `^(?<time>\d{2}-[A-Z]{1}[a-z]{2}-\d{4}\s\d{2}:\d{2}:\d{2}.\d{3})\s(?<level>[A-Z]+)\s\[(?<module>[^\]]+)\]\s(?<message>(?<source>[\w\.]+)[\S\s]+)`,
+					Parser: confgenerator.ParserShared{
+						TimeKey: "time",
+						//   13-Jan-2022 16:10:27.715
+						TimeFormat: "%d-%b-%Y %H:%M:%S.%L",
+						Types: map[string]string{
+							"lineNumber": "integer",
+						},
+					},
+				},
+			},
+		},
+		Rules: []confgenerator.MultilineRule{
+			{
+				StateName: "start_state",
+				NextState: "cont",
+				Regex:     `\d{2}-[A-Z]{1}[a-z]{2}-\d{4}\s\d{2}:\d{2}:\d{2}.\d{3}`,
+			},
+			{
+				StateName: "cont",
+				NextState: "cont",
+				Regex:     `^(?!\d{2}-[A-Z]{1}[a-z]{2}-\d{4}\s\d{2}:\d{2}:\d{2}.\d{3})`,
+			},
+		},
+	}.Components(tag, uid)
+
+	// https://tomcat.apache.org/tomcat-10.0-doc/logging.html
+	c = append(c,
+		fluentbit.TranslationComponents(tag, "level", "logging.googleapis.com/severity",
+			[]struct{ SrcVal, DestVal string }{
+				{"FINEST", "DEBUG"},
+				{"FINER", "DEBUG"},
+				{"FINE", "DEBUG"},
+				{"INFO", "INFO"},
+				{"WARNING", "WARNING"},
+				{"SEVERE", "CRITICAL"},
+			},
+		)...,
+	)
+	return c
+}
+
+type SystemLoggingReceiverTomcat struct {
+	LoggingProcessorTomcatSystem            `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+func (r SystemLoggingReceiverTomcat) Components(tag string) []fluentbit.Component {
+	if len(r.IncludePaths) == 0 {
+		r.IncludePaths = []string{
+			// Default log path on Ubuntu / Debian
+			"/opt/tomcat/logs/catalina.out",
+		}
+	}
+	c := r.LoggingReceiverFilesMixin.Components(tag)
+	c = append(c, r.LoggingProcessorTomcatSystem.Components(tag, "tomcat_system")...)
+	return c
+}
+
+type LoggingProcessorTomcatAccess struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (p LoggingProcessorTomcatAccess) Components(tag string, uid string) []fluentbit.Component {
+	return apacheAccessLogParser(tag, uid)
+}
+
+func (LoggingProcessorTomcatAccess) Type() string {
+	return "tomcat_access"
+}
+
+type AccessSystemLoggingReceiverTomcat struct {
+	LoggingProcessorTomcatAccess            `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+func (r AccessSystemLoggingReceiverTomcat) Components(tag string) []fluentbit.Component {
+	if len(r.IncludePaths) == 0 {
+		r.IncludePaths = []string{
+			// Default log path on Ubuntu / Debian
```

Comment:
What about the other distros? If they all share the same, we can remove this comment.

Reply:
removed.